libraries/mediaviewer/impl/src/main/kotlin/io/element/android/libraries/mediaviewer/impl/viewer/MediaViewerPresenter.kt

```diff
@@ -64,6 +72,37 @@ class MediaViewerPresenter @AssistedInject constructor(
         var currentIndex by remember { mutableIntStateOf(searchIndex(data, inputs.eventId)) }
         val snackbarMessage by snackbarDispatcher.collectSnackbarMessageAsState()
 
+        val isRenderingLoadingBackward by remember {
```

Comment:
This will cause the the Present method to recompose, I think you should try to manage this only using snapshot flows inside the **LaunchedEffect**. 
Maybe also extract that into a separate method.

Reply:
I have extracted to sub method. For the recomposition issue can we handle it later?